lib/whitehall_importer/integrity_checker.rb

```diff
@@ -95,28 +95,40 @@ def problem_description(message, expected, actual)
     end
 
     def primary_publishing_organisation_matches?
-      proposed_payload.dig("links", "primary_publishing_organisation") == publishing_api_link("primary_publishing_organisation")
+      proposed_payload.dig(:links, :primary_publishing_organisation) == publishing_api_link(:primary_publishing_organisation)
     end
 
     def organisations_match?
-      proposed_payload.dig("links", "organisations")&.sort == publishing_api_link("organisations")&.sort
+      proposed_payload.dig(:links, :organisations)&.sort == publishing_api_link(:organisations)&.sort
     end
 
     def publishing_api_content
       @publishing_api_content ||= if edition.live?
-                                    GdsApi.publishing_api.get_live_content(edition.content_id).to_h
+                                    GdsApi
+                                      .publishing_api
+                                      .get_live_content(edition.content_id)
+                                      .to_h
+                                      .deep_symbolize_keys
                                   else
-                                    GdsApi.publishing_api.get_content(edition.content_id).to_h
+                                    GdsApi
+                                      .publishing_api
+                                      .get_content(edition.content_id)
+                                      .to_h
+                                      .deep_symbolize_keys
                                   end
     end
 
     def publishing_api_link(link_type)
-      publishing_api_content.dig("links", link_type) ||
-        publishing_api_links.dig("links", link_type)
+      publishing_api_content.dig(:links, link_type) ||
+        publishing_api_links.dig(:links, link_type)
     end
 
     def publishing_api_links
-      @publishing_api_links ||= GdsApi.publishing_api.get_links(edition.content_id).to_h
+      @publishing_api_links ||= GdsApi
+        .publishing_api
+        .get_links(edition.content_id)
+        .to_h
+        .deep_symbolize_keys
```

Comment:
Minor: I wonder if this might benefit from having the method call on the same line and then inline. For example:

```
      @publishing_api_links ||= GdsApi.publishing_api
                                      .get_links(edition.content_id)
                                      .to_h
                                      .deep_symbolize_keys
```

Reply:
Sure - updated 👍